src/Symfony/Bundle/SecurityBundle/Security/FirewallContext.php

```diff
@@ -23,11 +23,25 @@ class FirewallContext
 {
     private $listeners;
     private $exceptionListener;
+    private $config;
 
-    public function __construct(array $listeners, ExceptionListener $exceptionListener = null)
+    public function __construct(array $listeners, ExceptionListener $exceptionListener = null, FirewallConfig $config = null)
     {
         $this->listeners = $listeners;
         $this->exceptionListener = $exceptionListener;
+
+        if (null === $config) {
+            @trigger_error(sprintf('"%s()" expects an instance of "%s" as third argument since version 3.2 and will throw an exception in 4.0 if not provided.', __METHOD__, FirewallConfig::class), E_USER_DEPRECATED);
```

Comment:
I got your point. Nothing depends on this argument, so there is no class that can't live without it. I will remove the deprecation.